domain/application/state/exposed.go

```diff
@@ -187,6 +193,37 @@ func (st *State) MergeExposeSettings(ctx context.Context, appID coreapplication.
 	return errors.Capture(err)
 }
 
+func (st *State) unsetAllExposedEndpoints(ctx context.Context, tx *sqlair.TX, appID coreapplication.ID) error {
```

Comment:
Why not just delete all the endpoints at once with an `IN`?

Reply:
Done, see 3447f80d5d326b630a8e23bb3e5f27d93ae88212.